services/job_chat/tests/test_pass_fail.py

```diff
@@ -0,0 +1,449 @@
+        processedAt: new Date().toISOString()
+      }
+    })
+  )
+);
+
+// Submit the final report
+post(
+  'https://reports.org/submit', 
+  state => ({
+    source: 'Patient API',
+    results: state.patients,
+    summary: state.summary,
+    endpoint: getPatientData('summary')
+  })
+);'''
+    }
+    
+    expected_code = '''// Custom helper functions
+function fetchPatientRecords(patientId) {
+  return `https://hospital.org/api/patients/${patientId}`;
+}
+
+function formatDate(date) {
+  return new Date(date).toISOString().split('T')[0];
+}
+
+// Initial configuration
+fn(state => {
+  const config = {
+    apiVersion: '2.1',
+    requestTimeout: 30000,
+    maxRetries: 3
+  };
+  return { ...state, config };
+});
+
+// Fetch data for each patient
+each(
+  '$.patients[*]',
+  fn(state => {
+    console.log(`Processing patient ID: ${state.data.id}`);
+    return state;
+  }),
+  get(state => fetchPatientRecords(state.data.id)),
+  fn(state => {
+    const patient = state.data;
+    // Enrich patient data
+    patient.lastUpdated = formatDate(new Date());
+    patient.dataSource = 'API';
+    patient.endpoint = fetchPatientRecords(patient.id);
+    
+    return { ...state, patient };
+  })
+);
+
+// Process the results
+fn(state => {
+  const patients = state.references.map(ref => ref.patient);
+  console.log(`Retrieved data for ${patients.length} patients`);
+  
+  const summary = {
+    count: patients.length,
+    source: fetchPatientRecords('summary'),
+    retrievalDate: formatDate(new Date())
+  };
+  
+  return { ...state, patients, summary };
+});
+
+// Save individual patient records
+each(
+  '$.patients[*]',
+  post(
+    'https://records.hospital.org/store',
+    state => ({
+      patient: state.data,
+      metadata: {
+        source: fetchPatientRecords(state.data.id),
+        processedAt: new Date().toISOString()
+      }
+    })
+  )
+);
+
+// Submit the final report
+post(
+  'https://reports.org/submit', 
+  state => ({
+    source: 'Patient API',
+    results: state.patients,
+    summary: state.summary,
+    endpoint: fetchPatientRecords('summary')
+  })
+);'''
+    
+    meta = {}
+    service_input = make_service_input(history=history, content=content, context=context, meta=meta, use_new_prompt=True)
+    response = call_job_chat_service(service_input)
+    print_response_details(response, "rename_function", content=content)
+    
+    assert response is not None
+    assert "response" in response
+    assert "suggested_code" in response
+    
+    assert response["suggested_code"] == expected_code, f"Function rename did not produce expected result.\nExpected:\n{expected_code}\n\nActual:\n{response['suggested_code']}"
+
+def test_change_multiple_instances():
+    print("==================TEST==================")
+    print("Description: Testing changing multiple instances of a URL path segment across longer job code.")
+    
+    history = []
+    content = "change all the occurences of endpoint to endpoooint"
```

Comment:
Doing this  the other way around, `endpoooint -> endpoint`, would feel like a more realistic usecase - using the model to fix an error.

What would be the consequences of having `endpoooint` in a string and expecting the model to ignore it? Like "rename all  variables" rather than "replace all values" is a more interesting and realistic AI test.

But a) maybe that's testing the wrong thing, and b) it introduces more scope for variation. What do you think?